presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -451,6 +453,13 @@ protected Type toPrestoType(int jdbcType)
 
     protected String toSqlType(Type type)
     {
+        if (type.equals(VARCHAR)) {
```

Comment:
Postgres limit for `VARCHAR` length is not strictly defined. I don't see a problem in letting connector interpret generic `VARCHAR` type in it's own way (using max length available) when creating table.